Teacher docs (#73)
dodona-edu/dodona-edu.github.io: nl/guides/exercise-series-management/README.md

---
title: Oefeningenreeksenbeheer
description: "Tutorial: oefeningenreeksenbeheer"
---

# Oefeningenreeksenbeheer

**Inhoudsopgave**
[[toc]]

Het [leerpad](/nl/course-manangement#leerpad) van een cursus bestaat uit verschillende [oefeningenreeksen](#oefeningenreeks) die elk opnieuw bestaan uit verschillenden oefeningen. Cursusbeheerders kunnen deze reeksen aanmaken, bewerken, verwijderen en herordenen. 

::: tip 

Al deze acties kunnen ook uitgevoerd worden op de oefeningen in een reeks. Deze kan je terugvinden in het [reeks-bewerken](#oefeningenreeks-bewerken) menu.

> :::

Author: A teacher can add, edit, delete and reorder series in the manage-series page. They can do these actions for exercises too when they scroll down in the edit-series page, and these actions work in a similar fashion (same icons etc)

::: tip Belangrijk

Als de deadline aangepast wordt dan krijgen cursusgebruikers meteen ook een indienstatus voor oefeningen uit de oefeningenreeks die aangepast is aan de nieuwe deadline.
:::

  Klik op het invulveld of klik op de kalenderknop om de datum en het tijdstip van de deadline in te stellen. Selecteer de deadline in de [tijdzone](../for-students#gebruikersprofiel-tijdzone) die je hebt ingesteld in je gebruikersprofiel. Andere gebruikers krijgen de deadline te zien in de tijdzone die ze in hun gebruikersprofiel hebben ingesteld.

  ![image](./staff.series_calendar_open.png)

  Klik op de verwijderknop om een ingestelde deadline te wissen.

  ![image](./staff.series_calendar_clear.png)

* `Zichtbaarheid`

  De zichtbaarheid bepaalt of gebruikers de oefeningenreeks kunnen zien. Voor deze eigenschap kunnen de volgende waarden ingesteld worden:

  * `Open`
   
    Alle gebruikers zien de oefeningenreeks in het leerpad.

  * `Verborgen`

>     Alleen cursusbeheerders zien de oefeningenreeks in het leerpad. Er staat een duidelijke mededeling bij om hen er op te wijzen dat andere gebruikers de oefeningenreeks enkel kunnen zien als ze de [geheime link](#oefeningenreeks-geheime-link) gebruiken. Er is ook een link aanwezig om deze instelling snel te veranderen.

Author: I don't think this is correct. Regular users cannot see the series unless they use the link. A slight error in the suggestion, I think.

![reeks bewerken opslaan](./staff.series_edit_submit.png)

Na het afwerken navigeer je terug naar de oefeningenreeks in het leerpad, waar je kunt vaststellen dat de nieuwe eigenschappen van de reeks meteen van toepassing zijn.

## Oefeningenreeks verwijderen

Uiteraard is het mogelijk om een reeks te verwijderen uit een cursus. De actie vind je analoog aan het bewerken in het reeksen-beheren menu of in het reeks-actiesmenu.

![reeks verwijderen](./staff.series_delete.png)

## Oefeningenreeks herordenen

Het kan handig zijn om reeksen in een cursus een bepaalde volgorde te geven, om ze bijvoorbeeld te sorteren volgens moeilijkheidsgraad. Standaard zullen ze gesorteerd worden in omgekeerd chronologische volgorde op basis van wanneer je ze toevoegt. Zo moeten studenten minder scrollen als ze een reeks willen maken. In het [reeksen-beherenpaneel](#oefeningenreeks-aanmaken) kan je in de tabel van de reeds toegevoegde reeksen ze verslepen via het icoontje aan de linkerkant.

## Oefeningenreeks beheren

> Dit menu bevat een aantal handige acties die cursusbeheerders kunnen uitvoeren op een reeks. Naast bewerken en verwijderen zijn er nog enkele mogelijkheden:

Reviewer: explain that the menu can be accessed by clicking on the 3 dots on the top right corner of a series

![reeks acties](./staff.series_actions_menu.png)

* `Reeks evalueren`

>   Deze actie stelt je in staat om op een gestructureerde manier de oplossingen van studenten voor de reeks te [evalueren](#oefeningenreeks-evalueren).

Reviewer: ```suggestion
  Deze actie stelt je in staat om op een gestructureerde manier door de ingediende oplossingen van deze reeks te bladeren, bijvoorbeeld om ze te [evalueren](#oefeningenreeks-evalueren) of verbeteren.
```

* `Statusoverzicht`

  Toont een overzicht met de indienstatus van alle cursusgebruikers voor alle oefeningen uit de oefeningenreeks. De indienstatus wordt in het overzicht weergegeven met de gebruikelijke [icoontjes](../for-students#oefeningenreeks-oefening-indienstatus-icoontje).

>  ![scoresheet](./staff.scoresheet.png)

Reviewer: screenshot shows an empty table without icons which is not really the best example.

  Klik op de naam van een cursusgebruiker om naar de [cursusoverzichtspagina](../user-management#cursusoverzichtspagina) van de gebruiker te navigeren.

>   ![scoresheet gebruikerslink](./staff.scoresheet_user_link.png)

Reviewer: remove screenshot

  Klik op het icoontje van een indienstatus om naar de oplossing te navigeren die gebruikt werd om de indienstatus te bepalen (als de cursusgebruiker effectief een oplossing  heeft ingediend op basis waarvan de indienstatus kon bepaald worden). Je kan in dit overzicht ook filteren op studenten die aan minstens één activiteit begonnen zijn en zoeken op naam.

>   ![scoresheet status](./staff.scoresheet_status_icon.png)

Reviewer: remove screenshot

* `Oplossingen van studenten exporteren`

>   Deze acties stelt je in staat om de ingezonden oplossingen van studenten voor de oefeningen in de reeks te [exporteren](#oefeningenreeks-oplossingen-exporteren).

Reviewer: ```suggestion
  Deze actie stelt je in staat om de ingezonden oplossingen van studenten voor de oefeningen in de reeks te [exporteren](#oefeningenreeks-oplossingen-exporteren) als zip-bestand.
```

* `Oplossingen hertesten`

>   Herevalueert alle oplossingen die cursusgebruikers ingediend hebben voor oefeningen van de oefeningenreeks. Dit kan nuttig zijn als een fout opgemerkt wordt in de automatische tests en de oplossingen opnieuw gecontroleerd moeten worden.

Reviewer: ```suggestion
  Deze actie hertest alle oplossingen die cursusgebruikers ingediend hebben voor oefeningen van de oefeningenreeks. Dit kan nuttig zijn als je bijvoorbeeld een aantal testen hebt toegevoegd of aangepast en je de al ingediende oplossingen opnieuw wil testen. In de volgende sectie wordt uitgelegd hoe je een enkele oplossing kan herevalueren.
```

> ## Oplossing herevalueren

Reviewer: ```suggestion
## Oplossing hertesten
```

> Bij het herevalueren van een oplossing wordt het [beoordelingsproces](../for-students#oplossing-beoordelingsproces) opnieuw uitgevoerd zonder dat de oplossing opnieuw moet ingediend worden. Op die manier blijft het originele tijdstip van indienen behouden. Als de configuratie van de oefening aangepast werd sinds de vorige beoordeling van de oplossing, dan kan de status van de oplossing wijzigen door het herevalueren.

Reviewer: text doesn't explain where this button is

Gebruikers krijgen geen melding van het platform als hun oplossingen geherevalueerd worden. Als je beslist om oplossingen te herevalueren, is het belangrijk om gebruikers te informeren dat er zowel wijzigingen kunnen zijn van de status van oplossingen die ze vroeger ingediend hebben als van hun indienstatus voor oefeningen in de oefeningenreeksen van de cursus.

:::

Klik op de herhaalknop in de rechterbovenhoek van de  [feedbackpagina](../for-students#feedbackpagina) om een oplossing te herevalueren.

![feedback evalueren](./staff.feedback_evaluate.png)

## Oefeningenreeks evalueren

Correcte testresultaten zijn geen garantie voor goede code. Daarom biedt Dodona ook ondersteuning om de oplossingen te evalueren en hen van feedback te voorzien. Om een evaluatie te starten, open je als lesgever het oefeningenreeks-actiesmenu.

![reeks-acties](./staff.series_actions_menu.png)

Vervolgens selecteer je `Reeks evalueren`.
![reeks evalueren](./staff.series_evaluate.png)

> Je wordt gebracht naar de volgende pagina waar je gebruikers kan selecteren om te evalueren.  Er zijn ook handige knoppen om snel een bepaalde categorie gebruikers te selecteren. Uiteraard kan je ook via de zoekbalk filteren op gebruikers via hun naam, labels, ... .

Reviewer: ```suggestion
Je word naar de volgende pagina gebracht waar je gebruikers kan selecteren om te evalueren.  Er zijn ook handige knoppen om snel een bepaalde categorie gebruikers te selecteren. Uiteraard kan je ook via de zoekbalk filteren op gebruikers via hun naam, labels, ... .
```

![select users](./staff.series_evaluate_select_users.png)

Na de nodige gebruikers gekozen te hebben, kan je de evaluate starten. 
![start evaluation](./staff.series_evaluate_start.png)

> Je wordt nu automatisch door de evaluatie geleid, waarbij Dodona bijhoudt welke gebruikers je reeds geëvalueerd hebt. Je kan de ingediende code van annotaties voorzien: opmerkingen over de code, wat goed is en wat beter kan. Deze feedback wordt niet automatisch vrijgegeven, dit kan je op het einde van de evaluatie voor alle gebruikers tegelijkertijd doen via de knop `Feedback vrijgeven`. Je kan deze feedback ook weer verbergen via dezelfde knop.

Reviewer: toevoegen dat studenten een melding krijgen als de feedback wordt vrijgegeven

Eronder vind je je voortgang voor de huidige oefening en kan je ook navigeren naar de oplossingen van de huidige gebruiker voor andere oefeningen die in de reeks zitten.
![andere oefeningen gebruiker](./staff.series_evaluate_feedback_row.png)

Via de navigatiebalk bovenaan kan je makkelijk terugkeren naar de evaluatiepagina.
![terugkeren](./staff.series_evaluate_return.png)

Je kan een reeks slechts één keer evalueren. Dit hoef je echter niet in één stuk te doen, je kan later terugkeren naar de evaluatie via het reeks-actiesmenu, waar nu `Evaluatie bekijken` staat.

![evaluatie bekijken](./staff.series_actions_check_evaluation.png)

Je kan een bestaande evaluatie ook verwijderen. De gegeven feedback zal ook verdwijnen.
![evaluatie verwijderen](./staff.series_evaluate_delete.png)

## Oefeningenreeks oplossingen exporteren

> In het actiesmenu van een reeks kan je als lesgever ook kiezen om de ingezonden code van je studenten te exporteren. Dit is handig als je liever op papier verbetert.

Reviewer: ```suggestion
In het actiesmenu van een reeks kan je als lesgever ook kiezen om de ingezonden code van je studenten te exporteren als een zip-bestand. Dit is bijvoorbeeld handig als je liever op papier verbetert en de code wil afdrukken.
```

![reeks exporteren](./staff.series_export_action.png)

Dit brengt je naar een exporteerpagina waar je eerst gevraagd wordt om de oefeningen in de reeks te selecteren waarvan je de inzendingen wenst. 
![oefeningen kiezen](./staff.series_export_exercise_choice.png)

Als je ze allemaal wenst te downloaden, dan kies je het bovenste selectievakje. Daarna klik je op `Volgende stap` om verder te gaan.

> ![volgende stap](./staff.series_export_exercises_chosen.png)

Reviewer: remove screenshot

Vervolgens kan je verschillende opties aanvinken die de inhoud van de export beïnvloeden. Je kan een samenvattende csv verkrijgen, kiezen of je alle oplossingen of enkel de laatste wil, of er rekening gehouden moet worden met de deadline, of de bestanden per student of per reeks gegroepeerd moeten worden en welke studenten meegerekend moeten worden. 

![export opties](./staff.series_export_options.png)

Klik op `Start export` om de download te starten.

> ![start export](./staff.series_export_start.png)

Reviewer: remove screenshot
